src/components/ShowList.tsx

```diff
@@ -0,0 +1,65 @@
+import IShow from "../utils/i-show";
+import { omitTags } from "../utils/omit-tags";
+import { useEffect, useState } from "react";
+
+interface ShowListProps {
+  setSelectedShow: React.Dispatch<React.SetStateAction<number | null>>;
+  show: IShow;
+}
+
+export default function ShowList({
+  setSelectedShow,
+  show,
+}: ShowListProps): JSX.Element {
+  const [isFavourite, setIsFavourite] = useState<boolean>(() => {
+    let value;
+    try {
+      value = JSON.parse(
+        window.localStorage.getItem(JSON.stringify(show.id)) || "false"
+      );
+    } catch (e) {
+      value = false;
+    }
+    return value;
+  });
+
+  useEffect(() => {
+    window.localStorage.setItem(
+      JSON.stringify(show.id),
+      JSON.stringify(isFavourite)
+    );
+  }, [isFavourite, show.id]);
+
+  return (
+    <div className="showEntry">
+      <h1 onClick={() => setSelectedShow(show.id)}>{show.name}</h1>
+      {isFavourite ? (
+        <button
+          onClick={() => {
+            setIsFavourite(false);
+          }}
+        >
+          ★
+        </button>
+      ) : (
+        <button
+          onClick={() => {
+            setIsFavourite(true);
```

Comment:
Like the way you used conditional rendering for the OnClick() and the setIsFavourite here. Pretty cool!